src/RaspiCapture.c

```diff
@@ -939,7 +939,7 @@ raspi_capture_fill_buffer(RASPIVID_STATE *state, GstBuffer **bufp,
 
   /* FIXME: Use our own interruptible cond wait: */
 
-  buffer = mmal_queue_timedwait(state->encoded_buffer_q, 500);
+  buffer = mmal_queue_timedwait(state->encoded_buffer_q, 10500);
```

Comment:
Looks like we need to make the timeout a bit better calculated. Arbitrarily increasing it to 10 seconds will annoy @staroselskii. I think the ideal will be something based on expected frame time + startup-delay-for-first-frame + 100ms or so, but for now just 500ms + when-variable-fps-add-1/shutter_speed might do